packages/mimir/src/rules/parameter-properties.ts

```diff
@@ -0,0 +1,193 @@
+                    for (const param of construct.parameters.filter(isParameterProperty))
+                        this.addFailure(
+                            construct.parameters.pos,
+                            construct.parameters.end,
+                            this.failureString,
+                            getFixerForDisallowedParameterProp(
+                                construct,
+                                param,
+                                getLineBreakStyle(this.context.sourceFile),
+                            ),
+                        );
+        }
+    }
+}
+
+function canBeParameterProperty(param: ts.ParameterDeclaration, construct: ts.ConstructorDeclaration): boolean {
+    const stmts = Array.from(construct.body!.statements);
+    if (getSuperCall(construct)) stmts.shift();
+    for (const stmt of stmts) {
+        /**
+         * See https://github.com/fimbullinter/wotan/issues/167#issuecomment-378945862 for details on this condition
+         */
+        if (
+            !isExpressionStatement(stmt) ||
+            !isBinaryExpression(stmt.expression) ||
+            !(stmt.expression.operatorToken.kind === ts.SyntaxKind.EqualsToken) ||
+            (isPropertyAccessExpression(stmt.expression.left) &&
+                stmt.expression.left.name.text === param.name.getText() &&
+                !isSimpleParamToPropAssignment(stmt, param))
+        )
+            return false;
+
+        if (isSimpleParamToPropAssignment(stmt, param)) return true;
+    }
+
+    return false;
+}
+
+function getFixerForDisallowedParameterProp(
+    construct: ts.ConstructorDeclaration,
+    param: ts.ParameterDeclaration,
+    lineBreak: string,
+): Replacement[] {
+    const superCall = getSuperCall(construct);
+    return [
+        /* Add assignment after super() call (if it exists) */
+        Replacement.append(
+            superCall ? superCall.end : construct.body!.getStart() + 1,
+            `${lineBreak}this.${param.name.getText()} = ${param.name.getText()};`,
+        ),
+
+        /* Add properties to class, trimming off default values if necessary */
+        Replacement.append(
+            construct.getStart(),
+            (param.getText().indexOf('=') > -1
+                ? param.getText().substring(0, param.getText().indexOf('=') - 1)
+                : param.getText()) +
+                ';' +
+                lineBreak,
+        ),
+
+        /* Finally, delete modifiers from the parameter prop */
+        Replacement.delete(param.pos, param.name.getStart()),
+    ];
+}
+
+function getFixerForLonghandProp(
+    param: ts.ParameterDeclaration,
+    construct: ts.ConstructorDeclaration,
+    classNode: ts.ClassDeclaration | ts.ClassExpression,
+): Replacement[] {
+    /* Class member and assignment to be removed */
+    const member = classNode.members
+        .filter(isPropertyDeclaration)
+        .find((prop) => prop.name.getText() === param.name.getText())!;
```

Comment:
this should also handle quoted property names:

```ts
class Foo {
  'bar': number;
  ['baz']: string;
  constructor(bar: number, baz: string) {
    this.bar = bar;
    this.baz = baz;
  }
}
```

you could use `tsutils.getPropertyName` to get the text of a statically known property name